packages/nft-playbook/src/app/cli/Commands/StartMintingCommand.ts

```diff
@@ -0,0 +1,111 @@
+import inquirer = require('inquirer');
+import fs = require('fs');
+import { Command, sleep } from './Command';
+
+import { deploy_contract } from '../../backend/deploy_contract';
+import { mint_nft } from '../../backend/mint_nft';
+import {
+  read_smart_contract,
+  read_user_transactions_on_smart_contract,
+} from '../../backend/read_smart_contract';
+
+let GAS_LIMIT;
+let server_uri;
+let priv_key_contract_owner;
+let priv_key_NFT_transmitter;
+let pub_key_NFT_receiver;
+
+export class StartMintingCommand implements Command {
+  name = 'Start Minting';
+  help = `\tStart the minting process`;
+
+  async execute() {
+    const promptQuestion: inquirer.QuestionCollection = [
+      {
+        type: 'input',
+        name: 'filename',
+        message: 'Please provide a settings.json',
+        default: './packages/nft-playbook/src/info.json',
+      },
+    ];
+    const answer = await inquirer.prompt(promptQuestion);
+
+    let file;
+    try {
+      file = fs.readFileSync(answer.filename, 'utf-8');
+    } catch (e) {
+      console.log('Please provide a valid filepath');
+      await sleep(5000);
+      return;
+    }
+
+    let info;
+    try {
+      info = JSON.parse(file);
+    } catch (error) {
+      console.log('Error parsing json file');
+      await sleep(5000);
+      return;
+    }
+
+    GAS_LIMIT = info.GAS_LIMIT;
+    server_uri = info.server_uri;
+    priv_key_contract_owner = info.priv_key_contract_owner;
+    priv_key_NFT_transmitter = info.priv_key_NFT_transmitter;
+    pub_key_NFT_receiver = info.pub_key_NFT_receiver;
+
+    if (
+      GAS_LIMIT === undefined ||
+      server_uri === undefined ||
+      priv_key_contract_owner === undefined ||
+      priv_key_NFT_transmitter === undefined ||
+      pub_key_NFT_receiver === undefined
+    ) {
+      console.log(
+        'Neccessary parameter missing. Please provide all required parameters in the json file.'
+      );
+      await sleep(5000);
+      return;
+    }
+
+    console.log(GAS_LIMIT);
+    console.log(server_uri);
+    console.log(priv_key_contract_owner);
+    console.log(priv_key_NFT_transmitter);
+    console.log(pub_key_NFT_receiver);
+
+    await sleep(5000);
+
+    // await this.mint();
+  }
+
+  async mint() {
+    let addr = undefined;
+    addr = await deploy_contract(
+      server_uri,
+      './packages/nft-playbook/src/app/backend/contracts/ERC721PresetMinterPauserAutoId.sol',
+      priv_key_contract_owner,
+      'NFT-DEMO',
+      '🚀',
+      'basis-uri'
+    );
+
+    console.log('Deployment successfull!');
+    console.log('Deployment successfull!');
+
+    await mint_nft(
+      server_uri,
+      priv_key_NFT_transmitter,
+      addr,
+      pub_key_NFT_receiver,
+      GAS_LIMIT
+    );
+
+    read_smart_contract(server_uri, addr);
+    read_user_transactions_on_smart_contract(
```

Comment:
I just thought about this from an understandability standpoint and it is very good for now, but I am not sure whether it should stay like this: 
We are currently deploying a contract in line 84, and then minting on that contract, and then reading that contract and the user transaction that happened on that contract. This is just like we merged it yesterday (Tuesday). Which is fine or even very good for this week and our current state. But it might become confusing due to the file name and the functionality it implies.

To increase the usability it might be good to separate those things a little during the next sprint (@lukaswolfrum & @Noah-Kurz), to that it is possible to explicitly deploy a contract (the addr could maybe be temporarily stored in an contract addr list and proposed when the minting process is started by the user?) and in a second step mint on that contract (And print only a confirmation instead of all of the contract & user related info => line 104 & 105)

Reply:
@all This is just an idea that we can discuss